File: FormulaBollo/src/shared/models/race.ts
import { Season } from "./season";

export class Race {
  id: number;
  name: string;
  flagImage: string;
  circuitImage: string;
  dateStart: Date;
  season: Season;
  finished: number;

  constructor(
    id: number,
    name: string,
    flagImage: string,
    circuitImage: string,
    dateStart: Date,
    season: Season,
    finished: number,
  ) {
    this.id = id;
    this.name = name;
    this.flagImage = flagImage;
    this.circuitImage = circuitImage;
    this.dateStart = dateStart;
    this.season = season;
    this.finished = finished;
  }
}

File: FormulaBolloApp/src/main/java/formula/bollo/entity/Race.java
package formula.bollo.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "race")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Race {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, length = 7)
    private Long id;

    @Column(name = "name", nullable = false, length = 50)
    private String name;

    @Column(name = "flag_image", nullable = false)
    private String flagImage;

    @Column(name = "circuit_image", nullable = false)
    private String circuitImage;

    @Column(name = "date_start", nullable = false)
    private LocalDateTime dateStart;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "season_id")
    private Season season;

    @Column(name = "finished", nullable = false, length = 1)
    private int finished;
}
